File: program_data/github_java_sort_function/10/38.java
public static double[] bucketSort(double[] array){
        
        double max = array[0];
        double min = array[0];
        for(int i=0;i<array.length;i++){
            if(array[i]>max){
                max = array[i];
            }
            if(array[i]<min){
                min = array[i];
            }
        }
        double d = max - min;
        
        
        int bucketNum = array.length;
        ArrayList<LinkedList<Double>> bucketList = new ArrayList<LinkedList<Double>>();
        for(int i=0;i<bucketNum;i++){
            bucketList.add(new LinkedList<Double>());
        }
        
        
        for(int i=0;i<array.length;i++){
            int num = (int)( (array[i]-min) * (bucketNum-1) / d);
            bucketList.get(num).add(array[i]);
        }

        
        for(int i=0;i<bucketList.size();i++){
            
            Collections.sort(bucketList.get(i));
        }
            
        
        double[] sortedArray = new double[array.length];
        int index = 0;
        for (LinkedList<Double> list : bucketList) {
            for (Double e : list) {
                sortedArray[index] = e;
                index++;
            }
        }
        
        return sortedArray;
    }

File: program_data/PKU_raw/76/38.c
void main()
{
	int n,a[50000],b[50000],c[10001]={0},i,j,max=0,min=10001,s=0;
	scanf("%d",&n);
	for(i=0;i<n;i++)
	{
		scanf("%d%d",&a[i],&b[i]);
		for(j=a[i];j<b[i];j++)
		{
			c[j]++;
		}
		if(a[i]<min)min=a[i];
		if(b[i]>max)max=b[i];
	}
	for(i=min;i<max;i++)
	{
		if(c[i]==0)
		{
			printf("no\n");
			s++;
			break;
		}
	}
	if(s==0)printf("%d %d\n",min,max);
}